### PR TITLE
Create draft schema based on the current on boarding workflow for Grasslands - Project Plan

### PR DESCRIPTION
**Pull request by blushi**
closes: https://github.com/regen-network/regen-registry/issues/378

The requirements are based on the following figma design: https://www.figma.com/file/a1A8H2i3gBzAq915IbfDAL/Seller-Flow, as well as this spreadsheet: https://docs.google.com/spreadsheets/d/1fAC1EY82tIzsiCEcXFU4klu58N5P0DI-DucAGzTmOu8/edit#gid=0

### CODE REVIEW

File: schema/project-plan.ttl [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+        sh:description "This is the name of the farm, ranch, property, or conservation project." ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+        sh:group regen:ProjectPlanBasicInfoGroup ;
+    ] ;
+    sh:property [
+        sh:path regen:size ;
+        sh:name "Size in hectares or acres" ;
+        rdfs:label "size";
```
Comment (mhagel) [inline schema/project-plan.ttl:39]:
```suggestion
        rdfs:label "size" ;
```
No idea if this is needed, but I noticed the rest have a space before the semicolon

File: schema/project-plan.ttl [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+    
+    sh:property [
+        sh:path regen:expectedLeakage ;
+        sh:name "Please indicate the expected leakage related to the proposed activities" ;
+        rdfs:label "expected leakage" ;
+        sh:group regen:ProjectPlanLeakageGroup ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+        sh:maxCount 1 ;
+        sh:in ("De Minimis" "Above De Minimis")
```
Comment (mhagel) [inline schema/project-plan.ttl:152]:
```suggestion
        sh:in ("De Minimis" "Above De Minimis") ;
```
ditto

File: schema/project-plan.ttl [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+		[
+			sh:property [
+                sh:path regen:permanenceRequirements ;
+                sh:hasValue "Allocate an additional 5% of each credit issuance (in addition to the Buffer Pool) to a dedicated Permanence Reversal Buffer."  ;
+            ] ;
+        ]
+		[
+			sh:property [
+                sh:path regen:permanenceRequirements ;
+                sh:hasValue "Register a covenant on the land from the Project Registration until the end of the 25-year permanence period."  ;
```
Comment (mhagel) [inline schema/project-plan.ttl:415]:
```suggestion
                sh:hasValue "Register a covenant on the land from the Project Registration until the end of the 25-year permanence period." ;
```

File: schema/project-plan.ttl [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+                    "Improved plant biodiversity"
+                    "Improved soil microbe diversity"
+                    "Improved water filtration"
+                    "Improved water quality"
+                    "Indigenous rights"
+                    "Reduced chemical runoff"
+                    "Reduced flooding"
+                    "Reduced soil erosion"
+                    "Social co-benefits"
+                )
```
Comment (mhagel) [inline schema/project-plan.ttl:538]: maybe need a semicolon here? IDK, I'm just looking for patterns at this point.
```suggestion
                ) ;
```

**Review by mhagel — APPROVED** [review]
I'm in favor of getting this in even though I cannot give critical feedback yet. I don't have the Shacl experience to be very helpful. I noticed a few pattern exceptions, I'm only noting them in case they were overlooked - maybe they are unimportant.

Thanks for being a trailblazer on this schema!

File: schema/project-plan.ttl [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+. 
+
+# Shape for projects that belong to biodiversity focused credit class
+regen:BiodiversityCreditClassShape
+    a rdfs:Class, sh:NodeShape ;
+    sh:targetClass regen:ProjectPlan ;
+    sh:property [
+        sh:path regen:plantSpecies ;
+        sh:name "Choose important plant species in this habitat" ;
+        rdfs:label "plant species";
```
Comment (haveanicedavid) [inline schema/project-plan.ttl:473]:
```suggestion
        rdfs:label "plant species" ;
```
(following Mark's lead)

**Review by haveanicedavid — APPROVED** [review]
This is clear to read, but my lack of contextual knowledge makes it hard to give a very thorough review. 

I think it'd be really helpful to go over this PR with all 3 of us, covering the syntax and how it'll be consumed by the registry app

File: schema/project-plan.ttl (hunk 4 above)
Comment (blushi) [reply schema/project-plan.ttl:538]: adding this one would make the file invalid as it's within an `sh:or`

**Comment by blushi** [comment]
> I think it'd be really helpful to go over this PR with all 3 of us, covering the syntax and how it'll be consumed by the registry app
Sure I'm actually working on the the last part (how it'll be consumed) as part of https://github.com/regen-network/regen-registry/issues/405, but we can review it all together during our next engineering checkin.

File: schema/project-plan.ttl [hunk 6]
```diff
@@ -0,0 +1,630 @@
...
+        sh:group regen:ProjectPlanEligibilityLandUseHistoryGroup ;
+    ] ;
+    sh:property [
+        sh:path regen:previousLandUse ;
+        sh:name "Prior to adopting these regenerative practices (going back 5 years prior) how were you using the land?" ;
+        rdfs:label "land use prior to regenerative practices" ;
+        sh:property [
+            sh:path ( [ sh:zeroOrMorePath rdf:rest ] rdf:first ) ;
+            sh:datatype xsd:string ;
+            sh:in ("Cropland" "Traditional Grazing" "Natural ecosystem, unmanaged") ;
```
Comment (aaronc) [inline schema/project-plan.ttl:60]: I think in general we should use IRIs for these enum type use cases.

```suggestion
            sh:in (regen:Cropland regen:TraditionalGrazing regen:NaturalEcosystemUnmanaged)
```

and then elsewhere define these IRIs and maybe assign them an `rdf:type` of `regen:LandUseType` or something...

File: schema/project-plan.ttl [hunk 7]
```diff
@@ -0,0 +1,630 @@
...
+        sh:name "Project name" ;
+        rdfs:label "project name" ;
+        sh:description "This is the name of the farm, ranch, property, or conservation project." ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+        sh:group regen:ProjectPlanBasicInfoGroup ;
+    ] ;
+    sh:property [
+        sh:path regen:size ;
```
Comment (mhagel) [inline schema/project-plan.ttl:37]: In addition to the actual size (a number), we will need to store whether the number refers to hectares or acres. For the front end, i tentatively called these `parcelSize` and `parcelUnit`.

Comment (aaronc) [reply schema/project-plan.ttl:37]: We likely want to adopt a unit ontology. Now there are several: QUDT, LINDT, OM, etc. There are some comparison articles we could look at https://www.google.com/search?q=rdf+unit+ontology+comparison&hl=en. The sparql community seemed to be leaning towards LINDT/UCUM: https://github.com/w3c/sparql-12/issues/129. But QUDT is also quite active: https://github.com/qudt/qudt-public-repo.

Comment (blushi) [reply schema/project-plan.ttl:37]: @mhagel This is handled by `ProjectSizeShape`

File: schema/project-plan.ttl (hunk 6 above)
Comment (blushi) [reply schema/project-plan.ttl:59]: makes sense, will update

File: schema/project-plan.ttl (hunk 7 above)
Comment (blushi) [reply schema/project-plan.ttl:39]: @aaronc I updated it to use QUDT, as it seemed at first glance more flexible to limit the accepted units, but we can revisit it if needed.

**Review by haveanicedavid — APPROVED** [review]

**Pull request merged**